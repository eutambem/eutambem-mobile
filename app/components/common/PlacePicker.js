import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Header } from 'react-navigation';

import { SafeAreaView, View, TextInput, StyleSheet, Text, Platform } from 'react-native';
import Modal from 'react-native-modal';
import { GooglePlacesAutocomplete } from 'react-native-google-places-autocomplete';

import appStyles from '../../styles';
import Button from '../common/Button';

const styles = StyleSheet.create({
  modalContainer: {
    flex: 1,
    justifyContent: 'flex-start',
    top: Header.HEIGHT,
  },
  innerContainer: {
    backgroundColor: 'white',
    opacity: 1,
    borderColor: 'rgba(0, 0, 0, 0.1)',
    justifyContent: 'flex-start',
    height: Platform.select({ android: '50%' }),
  },
  listView: {
    position: 'absolute',
    top: 44.5,
    backgroundColor: 'white',
  },
});

export default class PlacePicker extends Component {
  constructor(props) {
    super(props);
    this.state = {
      modalVisible: false,
      itemSelected: { value: '', label: '' },
    };
    this.onValueChange = this.onValueChange.bind(this);
  }

  onValueChange(item) {
    this.setState({ itemSelected: { value: item.place_id, label: item.description } });
    this.closeModal();
  }

  openModal() {
    this.input.blur();
    this.setState({ modalVisible: true });
  }

  closeModal() {
    this.setState({ modalVisible: false });
  }

  inputRef(component) { this.input = component; }

  render() {
    return (
      <SafeAreaView>
        <TextInput
          style={appStyles.input}
          placeholder={this.props.placeholder}
          value={this.state.itemSelected.label}
          onFocus={() => this.openModal()}
          ref={this.inputRef.bind(this)}
        />
        <Modal
          isVisible={this.state.modalVisible}
          style={styles.modalContainer}
          onRequestClose={() => this.closeModal()}
        >
          <SafeAreaView style={styles.innerContainer}>
            <View style={appStyles.modalHeader}>
              <Button
                type="cancel"
                onPress={() => this.closeModal()}
                title="Cancelar"
              />
              <Text style={{ fontSize: 18 }}>
                Selecione o Local
              </Text>
            </View>
            <GooglePlacesAutocomplete
              placeholder="Procurar"
              minLength={2}
              autoFocus={false}
              returnKeyType="search"
              listViewDisplayed="auto"
              fetchDetails
              renderDescription={row => row.description}
              onPress={this.onValueChange}
              getDefaultValue={() => this.state.itemSelected.label}
              query={{
                key: 'AIzaSyAWu91LURMU-W5OVonOPtPSKT94jJV50wQ',
                language: 'pt-BR',
                types: this.props.types,
              }}
              styles={{ listView: styles.listView }}
            />
          </SafeAreaView>
        </Modal>
      </SafeAreaView>
    );
  }
}

PlacePicker.propTypes = {
  placeholder: PropTypes.string.isRequired,
  types: PropTypes.arrayOf(PropTypes.string).isRequired,
};

