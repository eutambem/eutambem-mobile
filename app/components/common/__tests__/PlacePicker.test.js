import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';

import PlacePicker from '../PlacePicker';

const props = {
  types: ['establishment'],
  onPlaceChange: sinon.spy(),
};

describe('PlacePicker', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<PlacePicker {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
