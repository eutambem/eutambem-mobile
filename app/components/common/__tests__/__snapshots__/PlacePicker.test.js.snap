// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlacePicker renders correctly 1`] = `
<View>
  <Search
    autoFocus={true}
    backgroundColor="transparent"
    blurOnSubmit={true}
    cancelButtonTextStyle={
      Object {
        "color": "#8C8C8C",
      }
    }
    editable={true}
    inputStyle={
      Object {
        "padding": -10,
      }
    }
    keyboardShouldPersist={false}
    onChangeText={[Function]}
    placeholder="Procurar"
    placeholderCollapsedMargin={15}
    placeholderExpandedMargin={20}
    searchIconCollapsedMargin={30}
    searchIconExpandedMargin={10}
    shadowColor="#000"
    shadowOffsetHeightCollapsed={2}
    shadowOffsetHeightExpanded={4}
    shadowOffsetWidth={0}
    shadowOpacityCollapsed={0.12}
    shadowOpacityExpanded={0.24}
    shadowRadius={4}
    shadowVisible={false}
    useClearButton={true}
  />
  <FlatList
    data={Array []}
    disableVirtualization={false}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onEndReachedThreshold={2}
    renderItem={[Function]}
    scrollEventThrottle={50}
    updateCellsBatchingPeriod={50}
    windowSize={21}
  />
</View>
`;
